scripts/make_libmuscle_api.py

```diff
@@ -68,43 +66,44 @@ def __init__(self, with_names: bool) -> None:
                     instance_params.append(String(arg_name))
                 instance_name = 'grid_{}_n'.format(typ.tname())
-                fc_override=(
-                        'std::intptr_t LIBMUSCLE_$CLASSNAME$_create_grid_{0}_n_(\n'
-                        '        {1} * data_array,\n'
-                        '        std::size_t * data_array_shape,\n'
-                        '        std::size_t data_array_ndims,\n'
-                        '        char * index_name_1, std::size_t index_name_1_size,\n'
-                        '        char * index_name_2, std::size_t index_name_2_size,\n'
-                        '        char * index_name_3, std::size_t index_name_3_size,\n'
-                        '        char * index_name_4, std::size_t index_name_4_size,\n'
-                        '        char * index_name_5, std::size_t index_name_5_size,\n'
-                        '        char * index_name_6, std::size_t index_name_6_size,\n'
-                        '        char * index_name_7, std::size_t index_name_7_size\n'
-                        ') {{\n'
-                        '    std::vector<std::size_t> data_array_shape_v(\n'
-                        '            data_array_shape, data_array_shape + data_array_ndims);\n'
-                        '    auto data_array_p = const_cast<{1} const * const>(data_array);\n'
-                        '\n'
-                        '    std::vector<std::string> names_v;\n'
-                        '    names_v.push_back(std::string(index_name_1, index_name_1_size));\n'
-                        '    if (data_array_ndims >= 2u)\n'
-                        '        names_v.push_back(std::string(index_name_2, index_name_2_size));\n'
-                        '    if (data_array_ndims >= 3u)\n'
-                        '        names_v.push_back(std::string(index_name_3, index_name_3_size));\n'
-                        '    if (data_array_ndims >= 4u)\n'
-                        '        names_v.push_back(std::string(index_name_4, index_name_4_size));\n'
-                        '    if (data_array_ndims >= 5u)\n'
-                        '        names_v.push_back(std::string(index_name_5, index_name_5_size));\n'
-                        '    if (data_array_ndims >= 6u)\n'
-                        '        names_v.push_back(std::string(index_name_6, index_name_6_size));\n'
-                        '    if (data_array_ndims >= 7u)\n'
-                        '        names_v.push_back(std::string(index_name_7, index_name_7_size));\n'
-                        '\n'
-                        '    Data * result = new Data(Data::grid(\n'
-                        '            data_array_p, data_array_shape_v,\n'
-                        '            names_v, libmuscle::StorageOrder::first_adjacent));\n'
-                        '    return reinterpret_cast<std::intptr_t>(result);\n'
-                        '}}\n\n').format(typ.tname(), typ.fc_cpp_type())
+                fc_override = dedent("""\
```

Comment:
I don't mind reformatting like this, and with the dedent I think it's better than what was there, but could we make that a separate PR next time? It's very difficult to see what is a simple reformat and what is a change in functionality when they're mixed together.

Reply:
For example, I noticed the change from push_back to emplace_back in the generated code (nice!), but it took me forever to find where the generator had changed, because I read right over it when reviewing this file, and couldn't find it in the generator code itself (because it wasn't there, obviously).

Reply:
Yep, makes sense, sorry!